### PR TITLE
[Phar] use padraic/phar-updater for self-update

### PR DESCRIPTION
**Pull request by Naktibalda**
New implementation of phar self-update for Codeception 4.0
Uses [padraic/phar-updater](https://github.com/humbug/phar-updater)
Replaces #5745

### CODE REVIEW

File: src/Codeception/Command/SelfUpdate.php [hunk 1]
```diff
@@ -118,189 +106,15 @@ public function execute(InputInterface $input, OutputInterface $output)
...
-
-        if (isset($http_response_header)) {
-            $code = substr($http_response_header[0], 9, 3);
-            if (floor($code / 100) > 3) {
-                throw new \Exception($http_response_header[0]);
-            }
-        } else {
-            throw new \Exception('Request failed.');
+        return self::PHAR_URL;
+        if (version_compare(PHP_VERSION, '7.2.0', '<')) {
```
Comment (SamMousa) [inline src/Codeception/Command/SelfUpdate.php]: Unreachable code

Comment (Naktibalda) [reply src/Codeception/Command/SelfUpdate.php]: Oops, this is a bug. Thanks for noticing it.

Comment (Naktibalda) [reply src/Codeception/Command/SelfUpdate.php]: Fixed

**Comment by Naktibalda** [comment]
I merged #5751 to this branch to have all phar related changes in one place.

**Comment by Naktibalda** [comment]
https://github.com/Codeception/codeception.github.com/blob/e0b2603bac22f671d00439c57e215ce26b11bd28/releases/4.0.20191122/codecept.phar file self-updates to "current" 3.1.2 version.

**Pull request merged**